### PR TITLE
feat: Wallet v2 upgrade

### PR DESCRIPTION
**Pull request by TomMcL**
### Description

  - Moving to use wallet v2 transactions and key management
  - Switching from "multiple wallets" as the default setup for different parties to multiple keys within the same wallet. This makes switching within a console frontend easier.

### Testing

Integration tests updated to new format and working

### Breaking Changes

Where `wallet` used to be the necessary argument and `key` was optional, `key` is now the compulsory one with `wallet` generally optional

### CODE REVIEW

File: scripts/update_vega_version.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ VEGA_SIM_NETWORKS_TAG=master" > .env
 
 make
 make proto
-pytest -s -v --log-cli-level INFO -m integration
+# pytest -s -v --log-cli-level INFO -m integration
```
Comment (cdummett) [inline scripts/update_vega_version.sh]: Is this one intentional?

File: vega_sim/wallet/base.py [hunk 2]
```diff
@@ -48,7 +44,12 @@ def submit_transaction(self, transaction: Any, name: str, transaction_type: str)
...
         pass
 
-    def public_key(self, name: str) -> str:
+    @abstractmethod
+    def public_key(
+        self,
+        name: str,
+        wallet_name: Optional[str] = None,
+    ) -> str:
         """Return the public key associated with a given wallet name.
```
Comment (cdummett) [inline vega_sim/wallet/base.py]:
```suggestion
        """Return the public key associated with a given key name.
```

File: vega_sim/scenario/ideal_market_maker_v2/agents.py [hunk 3]
```diff
@@ -181,11 +179,12 @@ def initialise(self, vega: VegaServiceNull):
                 market_name=self.market_name,
                 proposal_wallet=self.wallet_name,
```
Comment (cdummett) [inline vega_sim/scenario/ideal_market_maker_v2/agents.py]: Think this arg needs to be removed.

```
(vega-sim-py3.10) Charlies-MacBook-Pro:vega-market-sim caed$ python -m vega_sim.scenario.adhoc -s historic_ideal_market_maker_v2 --console
INFO:vega_sim.null_service:Vega Running. Console launched at http://localhost:54756
INFO:vega_sim.null_service:Running NullChain from vegahome of /var/folders/mq/qntzn9jd0rngp7d0cj8nz9xh0000gn/T/vega-sim-4zu2fzfh
INFO:vega_sim.null_service:Launching GraphQL node at port 54750
INFO:vega_sim.null_service:Launching Console at port 54756
INFO:vega_sim.environment.environment:Running wallet at: http://localhost:54747
INFO:vega_sim.environment.environment:Running graphql at: http://localhost:54750
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/caed/vega-market-sim/vega_sim/scenario/adhoc.py", line 76, in <module>
    main()
  File "/Users/caed/vega-market-sim/vega_sim/scenario/adhoc.py", line 56, in main
    scenario.run_iteration(
  File "/Users/caed/vega-market-sim/vega_sim/scenario/scenario.py", line 79, in run_iteration
    outputs = self.env.run(
  File "/Users/caed/vega-market-sim/vega_sim/environment/environment.py", line 150, in run
    return self._run(self._vega, pause_at_completion=pause_at_completion)
  File "/Users/caed/vega-market-sim/vega_sim/environment/environment.py", line 192, in _run
    agent.initialise(vega=vega)
  File "/Users/caed/vega-market-sim/vega_sim/scenario/ideal_market_maker_v2/agents.py", line 178, in initialise
    self.vega.create_simple_market(
TypeError: VegaService.create_simple_market() got an unexpected keyword argument 'proposal_wallet'
```

**Review by cdummett — CHANGES_REQUESTED** [review]
Looks great! All the tests are passing for me!

I'm running into an issue though using the full vega wallet on nullchain:
 (changing `use_full_vega_wallet` to `True` in `adhoc.py`)

```
(vega-sim-py3.10) Charlies-MacBook-Pro:vega-market-sim caed$ python -m vega_sim.scenario.adhoc -s shaped_market_maker --console
INFO:vega_sim.null_service:Vega Running. Console launched at http://localhost:61728
INFO:vega_sim.null_service:Running NullChain from vegahome of /var/folders/mq/qntzn9jd0rngp7d0cj8nz9xh0000gn/T/vega-sim-2oxo_r29
INFO:vega_sim.null_service:Launching GraphQL node at port 61722
INFO:vega_sim.null_service:Launching Console at port 61728
INFO:vega_sim.environment.environment:Running wallet at: http://localhost:61719
INFO:vega_sim.environment.environment:Running graphql at: http://localhost:61722
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/caed/vega-market-sim/vega_sim/scenario/adhoc.py", line 76, in <module>
    main()
  File "/Users/caed/vega-market-sim/vega_sim/scenario/adhoc.py", line 56, in main
    scenario.run_iteration(
  File "/Users/caed/vega-market-sim/vega_sim/scenario/scenario.py", line 79, in run_iteration
    outputs = self.env.run(
  File "/Users/caed/vega-market-sim/vega_sim/environment/environment.py", line 150, in run
    return self._run(self._vega, pause_at_completion=pause_at_completion)
  File "/Users/caed/vega-market-sim/vega_sim/environment/environment.py", line 192, in _run
    agent.initialise(vega=vega)
  File "/Users/caed/vega-market-sim/vega_sim/scenario/common/agents.py", line 949, in initialise
    super().initialise(vega=vega, create_key=create_key)
  File "/Users/caed/vega-market-sim/vega_sim/environment/agent.py", line 80, in initialise
    self.vega.create_key(
  File "/Users/caed/vega-market-sim/vega_sim/service.py", line 309, in create_key
    return self.wallet.create_key(wallet_name=wallet_name, name=name)
  File "/Users/caed/vega-market-sim/vega_sim/wallet/vega_wallet.py", line 143, in create_key
    self.pub_keys.setdefault(wallet, {})[name] = json.loads(cmd)["publicKey"]
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
(vega-sim-py3.10) Charlies-MacBook-Pro:vega-market-sim caed$ 
```

Also what are the files in `vega_sim/data/wallets` and how are they showing in the changed files but not in my local branch?

File: vega_sim/environment/agent.py [hunk 4]
```diff
@@ -57,36 +56,31 @@ def __init__(
...
-        self.wallet_pass = wallet_pass
         self.key_name = key_name
 
         self.state_update_freq = state_update_freq
 
     def step(self):
         pass
 
-    def initialise(self, vega: VegaService, create_wallet: bool = True):
+    def initialise(self, vega: VegaService, create_key: bool = True):
```
Comment (cdummett) [inline vega_sim/environment/agent.py:77]: This change could be repeated in the parent method in `Agent`

Comment (cdummett) [inline vega_sim/environment/agent.py:77]: This change could be repeated in the parent method in `Agent` for clarity?

**Review by cdummett — CHANGES_REQUESTED** [review]
Feedback from running against network:

In `environment.py` agents are still being initialised with the `create_wallet` flag which is causing issues:

```
(vega-sim-py3.10) Charlies-MacBook-Pro:vega-market-sim caed$ python -m devops.run -s ETHUSD -n STAGNET1 --console
✓ Importing the network succeeded

Name:
stagnet1
File path:
/Users/caed/Library/Application Support/vega/wallet-service/networks/stagnet1.toml
INFO:vega_sim.network_service:Vega Running. Console launched at http://localhost:52388
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/caed/vega-market-sim/devops/run.py", line 92, in <module>
    main()
  File "/Users/caed/vega-market-sim/devops/run.py", line 82, in main
    scenario.run_iteration(
  File "/Users/caed/vega-market-sim/vega_sim/scenario/scenario.py", line 79, in run_iteration
    outputs = self.env.run(
  File "/Users/caed/vega-market-sim/vega_sim/environment/environment.py", line 415, in run
    return self._run(self._vega, pause_at_completion=pause_at_completion)
  File "/Users/caed/vega-market-sim/vega_sim/environment/environment.py", line 427, in _run
    agent.initialise(vega=vega, create_wallet=False, mint_wallet=False)
TypeError: ConfigurableMarketManager.initialise() got an unexpected keyword argument 'create_wallet'
```

File: scripts/update_vega_version.sh (hunk 1 above)
Comment (TomMcL) [reply scripts/update_vega_version.sh]: It is not :)

File: vega_sim/environment/agent.py (hunk 4 above)
Comment (TomMcL) [reply vega_sim/environment/agent.py:77]: :+1:

**Comment by TomMcL** [comment]
The files in `data/wallets` are deleted which is why they don't show up locally 😄. They're old default wallets from long ago which I don't think we want anymore

File: vega_sim/environment/agent.py [hunk 5]
```diff
@@ -21,7 +21,7 @@ def step(self, vega: VegaService):
         pass
 
     def initialise(
-        self, vega: VegaService, create_wallet: bool = False, mint_wallet: bool = False
+        self, vega: VegaService, create_key: bool = False, mint_wallet: bool = False
```
Comment (cdummett) [inline vega_sim/environment/agent.py]: I missed this first time, think `mint_wallet` should now be `mint_key` as well.

**Review by cdummett — CHANGES_REQUESTED** [review]
In the `_start_live_methods` in the `MarketEnvironment` class, the call to `vega.wallet.public_key` has the key and wallet arguments the wrong way round.

```
        party_ids = list(
            {
                vega.wallet.public_key(agent.wallet_name, agent.key_name)
                for agent in self.agents
                if hasattr(agent, "wallet_name")
            }
        )
```

should be

```
        party_ids = list(
            {
                vega.wallet.public_key(agent.key_name, agent.wallet_name)
                for agent in self.agents
                if hasattr(agent, "wallet_name")
            }
        )
```

**Review by cdummett — CHANGES_REQUESTED** [review]
One more change I forgot to add.

File: vega_sim/wallet/vega_wallet.py [hunk 6]
```diff
@@ -1,179 +1,283 @@
...
+            "params": [],
+            "id": "request",
+        }
+
+        url = f"{self.wallet_url}/api/v2/requests"
+
+        response = requests.post(url, headers=headers, json=submission)
         response.raise_for_status()
-        return {key["name"]: key["pub"] for key in response.json()["keys"]}
+        return {key["name"]: key["publicKey"] for key in response.json()["value"]}
```
Comment (cdummett) [inline vega_sim/wallet/vega_wallet.py]: This line isn't working for me running on a network.
```
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/caed/vega-market-sim/devops/run.py", line 92, in <module>
    main()
  File "/Users/caed/vega-market-sim/devops/run.py", line 82, in main
    scenario.run_iteration(
  File "/Users/caed/vega-market-sim/vega_sim/scenario/scenario.py", line 79, in run_iteration
    outputs = self.env.run(
  File "/Users/caed/vega-market-sim/vega_sim/environment/environment.py", line 415, in run
    return self._run(self._vega, pause_at_completion=pause_at_completion)
  File "/Users/caed/vega-market-sim/vega_sim/environment/environment.py", line 427, in _run
    agent.initialise(vega=vega, create_key=False, mint_key=False)
  File "/Users/caed/vega-market-sim/vega_sim/scenario/common/agents.py", line 1130, in initialise
    self.vega.submit_liquidity(
  File "/Users/caed/vega-market-sim/vega_sim/service.py", line 1354, in submit_liquidity
    return trading.submit_liquidity(
  File "/Users/caed/vega-market-sim/vega_sim/api/trading.py", line 381, in submit_liquidity
    wallet.submit_transaction(
  File "/Users/caed/vega-market-sim/vega_sim/wallet/vega_wallet.py", line 251, in submit_transaction
    pub_key = self.public_key(wallet_name=wallet_name, name=key_name)
  File "/Users/caed/vega-market-sim/vega_sim/wallet/vega_wallet.py", line 297, in public_key
    self.get_keypairs(wallet_name=wallet_name)
  File "/Users/caed/vega-market-sim/vega_sim/wallet/vega_wallet.py", line 236, in get_keypairs
    return {key["name"]: key["publicKey"] for key in response.json()["value"]}
KeyError: 'value'
```

Should it be:

```
return {key["name"]: key["publicKey"] for key in response.json()["result"]["keys"]}
```

**Review by cdummett — CHANGES_REQUESTED** [review]

**Review by cdummett — APPROVED** [review]

**Pull request merged**